src/include/libpmemfile-core.h

```diff
@@ -60,12 +60,19 @@ void pmemfile_pool_close(PMEMfilepool *pfp);
 
 PMEMfile *pmemfile_open(PMEMfilepool *pfp, const char *pathname, int flags,
 		...);
+PMEMfile *pmemfile_openat(PMEMfilepool *pfp, PMEMfile *file,
```

Comment:
I am not understanding why this is true.  One of the goals of the pmemfile core lib is to be easily usable for applications that want to use it as opposed to the interception layer. So,
pmemfile_openat() should look like:
int pmemfile_openat(PMEMfilepool *, int dirfd, const char *pathname, int flags);

There is no reason we cannot have the fd as part of the PMEMfile structure. This is the way I see it:
fd =open(LOGFILE, …);     ----->   fd =pmemfile_open(LOGFILE, …);
…
write(fd, buf, len);  ---> pmemfile_write(PMEMfielpool *, fd, buf, len);
…
fsync(fd); ---> no-op in pmemfile
...
close(fd) ---- > pmemfile_close(PMEMfilepool *, fd);

One of the mail goals is that the application developer 'understands' the language of the pmemfile core interface and can use the fd as they normally would in calls like pmemfile_openat().  When an open is done there is no reason, technically, that we cannot give them an fd back. The core library can open /dev/null and provide this, correct? What are the technical challenges with doing this?

Let's talk about this in our next meeting.